src/music/Scales.ts

```diff
@@ -0,0 +1,140 @@
+import { mod } from "../util/Util.js";
+
+import { Arrays } from "../util/Arrays.js";
+
+export type Scale = number; // scales are encoded with numbers in Ian Ring's format
+
+export interface ScaleQuery {
+    imperfections?: [number, number]; // allowed range for number of imperfections
+    hemitones?: [number, number]; // allowed range for number of hemitones
+    chord?: Scale; // chord that the scale must contain
+    notes?: [number, number]; // allowed range for number of notes in the scale
```

Comment:
Are these ranges inclusive or exclusive? They seem to be inclusive but you need to document that.